api/overdrive.py

```diff
@@ -106,19 +108,41 @@ class OverdriveAPI(BaseOverdriveAPI, BaseCirculationAPI, HasSelfTests):
     pdf = Representation.PDF_MEDIA_TYPE
     adobe_drm = DeliveryMechanism.ADOBE_DRM
     no_drm = DeliveryMechanism.NO_DRM
+    streaming_drm = DeliveryMechanism.STREAMING_DRM
     streaming_text = DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
-    overdrive_drm = DeliveryMechanism.OVERDRIVE_DRM
-
+    streaming_audio = DeliveryMechanism.STREAMING_AUDIO_CONTENT_TYPE
+    overdrive_audiobook_manifest = MediaTypes.OVERDRIVE_AUDIOBOOK_MANIFEST_MEDIA_TYPE
+    libby_drm = DeliveryMechanism.LIBBY_DRM
+
+    # These are not real Overdrive formats; we use them internally so
+    # we can distinguish between (e.g.) using "audiobook-overdrive"
+    # to get into Overdrive Read, and using it to get a link to a
+    # manifest file.
+    MANIFEST_INTERNAL_FORMATS = set(
+        ['audiobook-overdrive-manifest', 'ebook-overdrive-manifest']
+    )
+
+    # When a request comes in for a given DeliveryMechanism, what
+    # do we tell Overdrive?
     delivery_mechanism_to_internal_format = {
         (epub, no_drm): 'ebook-epub-open',
         (epub, adobe_drm): 'ebook-epub-adobe',
         (pdf, no_drm): 'ebook-pdf-open',
         (pdf, adobe_drm): 'ebook-pdf-adobe',
-        (streaming_text, overdrive_drm): 'ebook-overdrive',
+        (streaming_text, streaming_drm): 'ebook-overdrive',
+        (streaming_audio, streaming_drm): 'audiobook-overdrive',
+        (overdrive_audiobook_manifest, libby_drm): 'audiobook-overdrive-manifest'
     }
 
+    # Once you choose an Adobe format you're locked into it and can't
+    # use other formats.
+    LOCK_IN_FORMATS = [ x for x in BaseOverdriveAPI.FORMATS if 'adobe' in x ]
```

Comment:
Previously, we had a rule that once you chose a format, you were locked in, _unless _that was one of the the 'streaming' formats (e.g. Overdrive Read). But 'lock in' appears to be a feature specific to Adobe's ACS. So I flipped the logic: now only the formats with "adobe" in the name lock you in.